jans-auth-server/server/src/main/java/io/jans/as/server/service/RedirectionUriService.java

```diff
@@ -106,18 +106,19 @@ public String validateRedirectionUri(@NotNull Client client, String redirectionU
                 redirectUris = getSectorRedirectUris(sectorIdentifierUri).toArray(new String[0]);
             }
 
-            if (StringUtils.isNotBlank(redirectionUri) && redirectUris != null) {
+            if (StringUtils.isBlank(sectorIdentifierUri) && redirectUris != null && redirectUris.length == 1) {
+                return redirectUris[0];
+            }
+
+            if (StringUtils.isNotBlank(redirectionUri)) {
                 log.debug("Validating redirection URI: clientIdentifier = {}, redirectionUri = {}, found = {}",
                         client.getClientId(), redirectionUri, redirectUris.length);
-
                 if (isUriEqual(redirectionUri, redirectUris)) {
                     return redirectionUri;
                 }
-            } else {
-                // Accept Request Without redirect_uri when One Registered
-                if (redirectUris != null && redirectUris.length == 1) {
-                    return redirectUris[0];
-                }
+
+                //if not found a match, should it print a log ?
+                return redirectionUri.matches(client.getAttributes().getRedirectUrisRegex()) ? redirectionUri : null;
```

Comment:
Yes, we definitely has to print in log why we rejected call. Also if you can log other places where it is rejected it will be helpful.